packages/generate-object/src/index.ts

```diff
@@ -2,45 +2,71 @@ import type { GenerateTextOptions, GenerateTextResult } from '@xsai/generate-tex
 import type { Infer, InferIn, Schema } from 'xsschema'
 
 import { generateText } from '@xsai/generate-text'
-import { toJSONSchema, validate } from 'xsschema'
+import { toJSONSchema, validate, wrapArray, wrapObject } from 'xsschema'
 
 export interface GenerateObjectOptions<T extends Schema> extends GenerateTextOptions {
   schema: T
   schemaDescription?: string
   schemaName?: string
 }
 
-export interface GenerateObjectResult<T extends Schema> extends Omit<GenerateTextResult, 'text'> {
-  object: Infer<T>
-}
+type GenerateResult<O> = GenerateTextResult & { object: O }
+
+type OptionOutput = 'array' | 'object'
+
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T> & { output: 'array' }): Promise<GenerateResult<Array<Infer<T>>>>
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T> & { output: 'object' }): Promise<GenerateResult<Infer<T>>>
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T>): Promise<GenerateResult<Infer<T>>>
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T> & { output?: OptionOutput }) {
+  const { schema: rawSchema } = options
+
+  let wrappedSchema = rawSchema
+  if (options.output === 'array') {
+    wrappedSchema = await wrapObject(await wrapArray(options.schema), 'elements')
+  }
+
+  const schema = await toJSONSchema(wrappedSchema)
 
-/** @experimental WIP */
-export const generateObject = async <T extends Schema>(options: GenerateObjectOptions<T>): Promise<GenerateObjectResult<T>> =>
-  generateText({
+  const rawRet = await generateText({
     ...options,
     response_format: {
       json_schema: {
         description: options.schemaDescription,
         name: options.schemaName ?? 'json_schema',
-        schema: await toJSONSchema(options.schema),
+        schema,
         strict: true,
       },
       type: 'json_schema',
     },
-    schema: undefined,
-    schemaDescription: undefined,
-    schemaName: undefined,
+    schema,
+    schemaDescription: options.schemaDescription,
+    schemaName: options.schemaName,
```

Comment:
Revert them, this is to not have unwanted content in the request body

Reply:
in that case, just remove these three fileds would be better than keep them as undefined

Reply:
ok